Zend/zend_operators.c

```diff
@@ -742,6 +754,12 @@ ZEND_API zend_long ZEND_FASTCALL _zval_get_long_func(zval *op) /* {{{ */
 		case IS_DOUBLE:
 			return zend_dval_to_lval(Z_DVAL_P(op));
 		case IS_STRING:
+			if (!silent) {
+				if (is_numeric_string(Z_STRVAL_P(op), Z_STRLEN_P(op), NULL, NULL, -1)==0) {
+					zend_error(E_WARNING, "A non-numeric value encountered");
+					return 0;
+				}
+			}
```

Comment:
I think more consistent float support makes sense and worth the minor BC break in 7.1.